lightning/src/routing/network_graph.rs

```diff
@@ -543,6 +543,20 @@ impl NetworkGraph {
 	/// Scores each channel on the route by updating its channel_score instance within its
 	/// respective DirectionalChannelInfo.
 	pub fn score_channels_on_route(&mut self, route: Route, payment_succeeded: bool) -> Result<bool, LightningError> {
+		let paths = route.paths;
+		for path in paths {
+			for route_hop in path {
+				let short_channel_id = route_hop.short_channel_id;
+				let channel = self.channels.get_mut(&short_channel_id).unwrap();
```

Comment:
Also I don't think it's implemented right now but in the future you may learn a payment success _after_ a channel closure onchain, so it would be better to unwrap the None here and return a processing error.

Reply:
In the event of unwrapping a None, would it make sense to exit the function and return a LightningError and stopping the scoring of the rest of the channels?

Reply:
No, I don't think so because MPP ?

1) You learn about a network-wise channel failure
2) You learn about a partial-failure among your MPP-route, due to this channel failure
3) You skip scoring for this channel but do want to reward well-behaving ones.

Reply:
Ah ok, makes sense. Instead of returning an error there I'm just going to let the method continue with awarding the other channels that are still active. The method will return an error indicating that not all channels were scored after it has gone through every RouteHop on the route.